helm/oauth2-proxy/templates/secret-provider-class.yaml

```diff
@@ -0,0 +1,12 @@
+{{- if .Values.secretProviderClass }}
```

Comment:
To me make sense to use the same approach of other k8s object
Ref: https://github.com/oauth2-proxy/manifests/blob/fb57298898d94fc3491e0b4c6fa695a63d6f2111/helm/oauth2-proxy/values.yaml#L104

Reply:
Reason I didn't go that route is if content is specified in the secretProviderClass then it can be assumed that it's enabled. This is similar to what is done in the kube-prometheus-stack helm chart - https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/templates/prometheus/csi-secret.yaml